### DISCUSSION TITLE
How to Close Sidebar on Item Click (mobile) ?

### DISCUSSION THREAD

**Discussion by reksamamur** [unovue/shadcn-vue#960] [Q&A]
Hi, hopping someone can help.

I can't seem to find work around on it, tried using `useSidebar()` but got error injection.

**Comment by drewtownchi** [comment]
What I ended up doing was adding the following to Sidebar.vue

```js
defineExpose({
  isMobile,
  setOpenMobile,
})
```

and then on my layout page 

```html
    <Sidebar ref="sidebarRef">
   <!-- other code -->
                <NuxtLink :to="item.href" @click="sidebarRef?.isMobile ? sidebarRef?.setOpenMobile(false) : null">
                  <Icon :name="item.icon" class="flex-shrink-0 w-5 h-5 mr-2" /> {{ item.name }}
                </NuxtLink>
```

**Reply by adlermore** [reply]
What I ended up doing was adding the following to Sidebar.vue:

`defineExpose({
  isMobile,
  setOpenMobile,
})`
Then on my layout page:

`<Sidebar ref="sidebarRef">
  
  <NuxtLink :to="item.href" @click="sidebarRef?.isMobile ? sidebarRef?.setOpenMobile(false) : null">
    <Icon :name="item.icon" class="flex-shrink-0 w-5 h-5 mr-2" /> {{ item.name }}
  </NuxtLink>
`

This approach worked for me to handle mobile behavior in the sidebar when a link is clicked.

**Comment by jordihoven** [comment]
Hi, i just started a discussion on this topic and found your question. Have a look at [my discussion](https://github.com/unovue/shadcn-vue/discussions/1604).
